image-engine/src/main/java/com/safehill/kcrypto/image_engine/internal/MLModelsRepositoryImpl.kt

```diff
@@ -0,0 +1,132 @@
+package com.safehill.kcrypto.image_engine.internal
+
+import android.content.Context
+import android.os.SystemClock
+import com.safehill.kcrypto.image_engine.model.DownloadModelState
+import com.safehill.kcrypto.image_engine.model.MLModel
+import com.safehill.kcrypto.image_engine.MLModelsRepository
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.cio.CIO
+import io.ktor.client.plugins.HttpTimeout
+import io.ktor.client.request.prepareGet
+import io.ktor.client.statement.bodyAsChannel
+import io.ktor.http.contentLength
+import io.ktor.http.isSuccess
+import io.ktor.utils.io.core.isEmpty
+import io.ktor.utils.io.core.readAvailable
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.channelFlow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import java.io.File
+import java.util.UUID
+
+// TODO: improve temp file handling, currently there is no logic to delete them when app crashes
```

Comment:
Why a temp file? Why deleting?

Reply:
I think it's better to create a temporary file to store the downloaded data and then move it to the proper path to avoid accessing a incomplete version of the file